pkg/mtu/detect_linux.go

```diff
@@ -58,16 +58,18 @@ func autoDetect() (int, error) {
 			return 0, fmt.Errorf("%v, %v", err.Error(), prevErr.Error())
 		}
 	}
+	if routes[0].Gw != nil {
+		link, err := netlink.LinkByIndex(routes[0].LinkIndex)
+		if err != nil {
+			return 0, fmt.Errorf("unable to find interface of default route: %s", err)
+		}
 
-	link, err := netlink.LinkByIndex(routes[0].LinkIndex)
-	if err != nil {
-		return 0, fmt.Errorf("unable to find interface of default route: %s", err)
-	}
-
-	if mtu := link.Attrs().MTU; mtu != 0 {
-		log.Infof("Detected MTU %d", mtu)
-		return mtu, nil
+		if mtu := link.Attrs().MTU; mtu != 0 {
+			log.Infof("Detected MTU %d", mtu)
+			return mtu, nil
+		}
+		return EthernetMTU, nil
+	} else {
+		return 0, fmt.Errorf("unable to find default gateway from routes: %s", routes)
```

Comment:
Nit: Might be nicer to implement this as an early check, to avoid one more indentation:
```go
if routes[0].Gw == nil {
    return 0, fmt.Errorf("unable to find default gateway from routes: %s", routes)
}
```

Reply:
Done.